src/SAWScript/Crucible/LLVM/Builtins.hs

```diff
@@ -469,10 +470,44 @@ verifyPrestate opts cc mspec globals = do
             (mspec ^. MS.csPreState . MS.csFreshPointers)
   let env = Map.unions [env1, env2]
 
-  mem'' <- setupPrePointsTos mspec opts cc env (mspec ^. MS.csPreState . MS.csPointsTos) mem'
-  let globals1 = Crucible.insertGlobal lvar mem'' globals
-  (globals2,cs) <- setupPrestateConditions mspec cc env globals1 (mspec ^. MS.csPreState . MS.csConditions)
-  args <- resolveArguments cc mspec env
+  mem'' <-
+    foldM (\m (LLVMAllocGlobal _ symbol@(L.Symbol name)) -> do
```

Comment:
This big lambda should be factored out into a separate function with a comment explaining what it does